pkg/controllers/management/auth/prtb_sa_controller.go

```diff
@@ -0,0 +1,52 @@
+package auth
+
+import (
+	apiv3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	v3 "github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/types/config"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	prtbServiceAccountControllerName = "prtb-service-account-controller"
+	serviceAccountAnnotation         = "serviceAccount"
+)
+
+type prtbServiceAccountController struct {
+	prtbLister v3.ProjectRoleTemplateBindingLister
+	prtbClient v3.ProjectRoleTemplateBindingInterface
+}
+
+func newPRTBServiceAccountController(mgmt *config.ManagementContext) *prtbServiceAccountController {
+	return &prtbServiceAccountController{
+		prtbLister: mgmt.Management.ProjectRoleTemplateBindings("").Controller().Lister(),
+		prtbClient: mgmt.Management.ProjectRoleTemplateBindings(""),
+	}
+}
+
+func (c prtbServiceAccountController) sync(_ string, obj *apiv3.ProjectRoleTemplateBinding) (runtime.Object, error) {
+	if obj == nil {
+		return obj, nil
+	}
+	prtbs, err := c.prtbLister.List("", labels.Everything())
+	if err != nil {
+		return nil, err
+	}
+	for _, prtb := range prtbs {
```

Comment:
We definitely shouldn't be listing every PRTB in this function. In the current format, this will list/evaluate every PRTB every single time even a single one is changed. This will likely cause a large performance hit in production, and it's unecessary since this function really should only be evaluating a single PRTB (the one passed in as a parameter).